### PR TITLE
[HEAP-11501] Use iOS Framework track methods

### PR DESCRIPTION
**Pull request by jmtaber129**
## Description
* Upgrade the vendored iOS Heap dependency to 6.5.0-alpha.1
* Switch the native iOS methods the bridge uses from custom track to the framework track methods provided in Heap iOS v6.5.0
* Change the `type` field in navigation events to `action`, since `type` is reserved for the event name
* Update tests + test utils to use `source` and `sprops` pixel params.

This will introduce inconsistency between how iOS and Android events are sent and processed server-side, so merging into `feature/rnfcl` for now.

## Test Plan
Updated + ran detox tests, and observed correct events coming to kafka.

## Checklist
- [X] Detox tests pass (only Heap employees are able run these)
- [ ] ~If this is a bugfix/feature, the changelog has been updated~ (will be updated right before `feature/rnfcl` is merged into develop)

### CODE REVIEW

File: js/autotrack/reactNavigation.js [hunk 1]
```diff
@@ -29,7 +29,7 @@ export const withReactNavigationAutotrack = track => AppContainer => {
 
       track(EVENT_TYPE, {
         path: initialPageviewPath,
-        type: INITIAL_ROUTE_TYPE,
```
Comment (GrandPoohBear) [inline js/autotrack/reactNavigation.js]: Remind me - what's the motivation for this change?  Consistency with something else, or is `type` a special keyword somewhere?

Comment (jmtaber129) [reply js/autotrack/reactNavigation.js]: `type` represents the event type (param `t` in the pixel).  So if we pick a `type` field from `sprops` and throw it into the non-custom part of the event getting written to kafka, this field would overwrite the *real* `type`.  This used to not be an issue, since `type` was treated as a custom prop.
